### PR TITLE
Bugfix/680/fix examples with animation

### LINKED ISSUE #680: Animation example doesn't yet work after Specs refactor
The animation example has stopped working after the Specs refactor, running `cargo +nightly --example animation` with `shred`'s `nightly` feature turned on gives this:

```
thread 'main' panicked at 'Tried to fetch a resource of type "amethyst::<unnamed>::storage::MaskedStorage<amethyst::<unnamed>::AnimationSet<AnimationId, amethyst::<unnamed>::Transform>>", but the resource does not exist.
Try adding the resource by inserting it manually or using the `setup` method.', /home/azriel/.cargo/registry/src/github.com-1ecc6299db9ec823/shred-0.7.0-alpha5/src/res/mod.rs:215:48
```

(gotta love that panic message :100:)

The [failing line](https://github.com/amethyst/amethyst/blob/c7b58f2cb37a8c8df34fe7024629c5b24a8a265b/examples/animation/main.rs#L354) is `world.write_storage().insert(entity, set);`, i.e. it can't get the `MaskedStorage<AnimationSet<_, _>>`.

I haven't investigated further than this yet; I have to go somewhere now.

---

Stack backtrace:

```
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:205
   3: std::panicking::default_hook
             at libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:457
   5: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:344
   6: shred::res::Resources::fetch_mut::{{closure}}
             at ./<panic macros>:8
   7: <core::option::Option<T>>::unwrap_or_else
             at /checkout/src/libcore/option.rs:376
   8: shred::res::Resources::fetch_mut
             at /home/azriel/.cargo/registry/src/github.com-1ecc6299db9ec823/shred-0.7.0-alpha5/src/res/mod.rs:215
   9: specs::storage::data::<impl shred::system::SystemData<'a> for specs::storage::Storage<'a, T, shred::res::FetchMut<'a, specs::storage::MaskedStorage<T>>>>::fetch
             at /home/azriel/.cargo/registry/src/github.com-1ecc6299db9ec823/specs-0.11.0-alpha4/src/storage/data.rs:206
  10: specs::world::World::write_storage
             at /home/azriel/.cargo/registry/src/github.com-1ecc6299db9ec823/specs-0.11.0-alpha4/src/world/mod.rs:311
  11: animation::initialise_animation
             at examples/animation/main.rs:354
  12: <animation::Example as amethyst::state::State>::on_start
             at examples/animation/main.rs:60
  13: amethyst::state::StateMachine::start
             at src/state.rs:82
  14: amethyst::app::Application::initialize
             at src/app.rs:151
  15: amethyst::app::Application::run
             at src/app.rs:127
  16: animation::run
             at examples/animation/main.rs:210
  17: animation::main
             at examples/animation/main.rs:215
  18: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  19: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:304
  20: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:105
  21: std::rt::lang_start_internal
             at libstd/panicking.rs:283
             at libstd/panic.rs:361
             at libstd/rt.rs:58
  22: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  23: main
  24: __libc_start_main
  25: _start
```

### PR DESCRIPTION
**Pull request by azriel91**
Fixes #680.

I've tested the following examples to make sure their animations work:

* animation
* gltf
* sprites

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/681)

### CODE REVIEW

File: amethyst_animation/src/systems/control.rs [hunk 1]
```diff
@@ -194,6 +194,14 @@ where
             controls.remove(entity);
         }
     }
+
+    fn setup(&mut self, res: &mut Resources) {
```
Comment (torkleyy) [inline amethyst_animation/src/systems/control.rs]: Always call `Self::SystemData::setup`, too.

Comment (azriel91) [reply amethyst_animation/src/systems/control.rs]: haha @Rhuagh just mentioned that on the chat, fixed!

File: amethyst_animation/src/systems/control.rs [hunk 2]
```diff
@@ -194,6 +194,12 @@ where
...
+        Self::SystemData::setup(res);
+        ReadStorage::<AnimationSet<I, T>>::setup(res);
+        ReadStorage::<Handle<Animation<T>>>::setup(res);
```
Comment (torkleyy) [inline amethyst_animation/src/systems/control.rs]: Shouldn't this get added by the processor?

Comment (Rhuagh) [reply amethyst_animation/src/systems/control.rs]: Already discussed. The Animaton handle should not be used as a component.

**Comment by torkleyy** [comment]
Guess there aren't many who could object to this.

bors r+

**Pull request merged**